deploy/libproject/deploy.rb

```diff
@@ -63,23 +131,29 @@ def deploy(cmd_name, args)
     "default service account."
   )
   op.add_option(
-    "--key_file [key file]",
+    "--key-file [key file]",
     lambda {|opts, v| opts.key_file = v},
     "Path to a service account key file to be used for deployment"
   )
   op.add_option(
-    "--git_version [git version]",
+    "--git-version [git version]",
     lambda {|opts, v| opts.git_version = v},
     "GitHub tag or branch, e.g. 'v1-0-rc1', 'origin/master'. Branch names " +
     "must be prefixed with 'origin/'. By default, uses the current live " +
     "staging release tag (if staging is in a good state)"
   )
   op.add_option(
-    "--app_version [app version]",
+    "--app-version [app version]",
     lambda {|opts, v| opts.app_version = v},
     "App Engine version to deploy as. By default, uses the current live " +
     "staging release version (if staging is in a good state)"
   )
+  op.add_option(
+    "--no-update-jira",
```

Comment:
The reason I wanted an escape hatch is because doing a JIRA update adds more constraints:

- The version you are pushing looks like a release version, e.g v0-1-rc1
- Staging must be in a consistent state, UI, public-api, and api all have the same version; that version looks like a release version

(See validation at 210)

The only common scenario I expect to need this is if we need to redeploy because staging is in a bad state, e.g. a staging deploy failed halfway through. Then we wouldn't be able to generate a commit delta and therefore couldn't create a ticket.

Reply:
Also, I don't think there are any cases where we *wouldn't* want to update JIRA for stable/prod. But currently it is nearly impossible to hack in local changes to this deploy script without first merging and including them in your release (this may wind up being an issue - I'll probably try to make that better in the future), so I figured I should allow flexibility with flags where possible.